subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -56,7 +56,7 @@
     /**
      * Executes the work synchronously.
      */
-    WorkOutput execute(@Nullable InputChangesInternal inputChanges, InputChangesContext context);
+    WorkOutput execute(File workspace, @Nullable InputChangesInternal inputChanges, @Nullable ImmutableSortedMap<String, FileSystemSnapshot> previousOutputs);
```

Comment:
I think it would be better to pass in an interface to this method, so it is easier to add things to the object we pass in, should we need some more data in some places. How about passing in an interface `ExecutionContext`, which is part of the "public" api of the execution engine?

Reply:
Sounds like a good idea. It would allow us to use `Optional` instead of `@Nullable`, too.